feat(docs): add introduction to the dev docs website (#75)
coffee-tools/coffee: docs/docs-book/src/introduction.md

# What is Coffee? and Why do we need it?

> Coffee is a plugin manager for core lightning that supports the user
> and the developers while working and using the core lightning plugin
> architecture.
> 
> > The plugin manager is under development so at the current moment
> only the code is the real docs and some of the things that are marked as
> unsupported in the docs may be already supported in the code

Reviewer: ```suggestion
The Coffee plugin manager facilitates the use of the Core Lightning plugin architecture for both users and developers. It offers assistance to them while working with Core Lightning.

> At this time, the only true documentation for the plugin manager is the code itself, and some features that are listed as unsupported in the documentation may already be supported in the code, as the plugin manager is still in the process of being developed.
```

## Problems

Core lightning has a very powerful plugin system that empowers
everyone to write a plugin for core lightning to extend the functionality
of it.

In fact, some of the core features like the `pay` command in core lightning is
itself a plugin.

> However, usually when you give the power to everyone to write their own
> way to write a plugin, this will start to fragment the ecosystem in a very
> bad way.
> 

Reviewer: ```suggestion
Despite allowing all users to write plugins in their own way, this can have a detrimental effect in fragmenting the ecosystem.

```

> In fact, currently, we had many ways to write a plugin in core lightning
> with different languages, like rust, go, java, kotlin, scala, dart, python

Reviewer: ```suggestion
In fact, currently, we have many ways to write a plugin in core lightning
with different languages, like Rust, Go, Java, Kotlin, Scala, Dart, Python
```

> and many others. And this can cause confusion around the community and
> the user now has a different way to install a plugin and different
> way to compile and install a plugin.

Reviewer: ```suggestion
and many others. This can lead to disarray in the community, as the user now has to follow a different procedure for installing a plugin and building it.
```

## Solution

> Our solution that we want to provide with coffee is a single reference to
> install a plugin and develop a plugin for core lightning, as cargo is for Rust.

Reviewer: ```suggestion
We are offering Coffee as a solution to help install and develop a plugin for Core Lightning, similar to how Cargo is used for Rust.
```

> In fact, coffee is implemented to be a modular plugin manager and supports all

Reviewer: ```suggestion
In fact, Coffee is implemented to be a modular plugin manager and supports all
```

the user needs.

> Currently, with coffee it is possible to install a plugin from a github repository

Reviewer: ```suggestion
Currently, with Coffee it is possible to install a plugin from a GitHub repository
```

and allow the user to manage it.

> In fact, you can install a java plugin that supports coffee in a couple of

Reviewer: ```suggestion
In fact, you can install a Java plugin that supports Coffee in a couple of
```

commands after you [install it](./install-coffee.md)

```bash
coffee --network testnet setup /home/alice/.lightning
coffee --network testnet remote add lightningd https://github.com/lightningd/plugins.git
coffee --network install btcli4j
```

>The plugin manager is currently under development, so some of the plugins can
be not supported, but please submit an issue
[here](https://github.com/coffee-tools/coffee/issues) and we try to help
you :smile:

## Ideas

> The idea that we had for coffee is to be the facto tool to install and

Reviewer: ```suggestion
The idea that we have for Coffee is to be the de facto tool to install and
```

> build a plugin in core lightning, so we are planning to release it in different

Reviewer: ```suggestion
build a plugin in Core Lightning, so we are planning to release it in different
```

way, such as:

> - command line: empower the developers and the Linux hackers to play with coffee from command line;

Reviewer: ```suggestion
- Command line: empower the developers and the Linux hackers to play with Coffee from command line;
```

- webserver: empower the web application and mobile application to manage core lightning
plugins;

> - plugins: empower current systems like grpc and rest API to interact with

Reviewer: ```suggestion
- Plugins: empower current systems like gRPC and REST to interact with
```

> with coffee through a core lightning RPC commands.

Reviewer: ```suggestion
with Coffee through the Core Lightning RPC commands.
```
